ApiBundle/Controller/NodeController.php

```diff
@@ -117,18 +116,10 @@ public function duplicateAction(Request $request, $nodeId)
         $language = $request->get('language');
         $siteId = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteId();
         /** @var NodeInterface $node */
-        $node = $this->get('open_orchestra_model.repository.node')
-            ->findOneByNodeIdAndLanguageAndVersionAndSiteId($nodeId, $language, $siteId);
-        $newNode = $this->get('open_orchestra_backoffice.manager.node')->duplicateNode($node);
-
-        $this->dispatchEvent(NodeEvents::NODE_DUPLICATE, new NodeEvent($newNode));
-
-        $em = $this->get('doctrine.odm.mongodb.document_manager');
-        $em->persist($newNode);
-
-        $this->get('open_orchestra_backoffice.manager.node')->updateBlockReferences($node, $newNode);
-
-        $em->flush();
+        $documentManager = $this->get('doctrine.odm.mongodb.document_manager');
+        $documentManager->getConnection()->initialize();
+        $dataBase = $documentManager->getDocumentDatabase('OpenOrchestra\ModelBundle\Document\Node');
+        $dataBase->execute("db.loadServerScripts();duplicateNode({ nodeId: 'root', siteId: '2', language: 'en' });");
```

Comment:
yes, you could test it on your vagrant first. @itkg-alavieille does jmeter work on the vagrant (maybe reduce the number of users)